File: c++/lecture/class_test/account/entity/Account.cpp
//
// Created by eddi on 23. 12. 5.
//

#include "Account.h"

Account::Account(const std::string& account_id, const std::string& password)
                        : account_id(account_id), password(password) { }

Account::Account(int id, const std::string& account_id, const std::string& password) : id(id),
                                                                                       account_id(account_id),
                                                                                       password(password),
                                                                                       reg_date(std::chrono::system_clock::now()),
                                                                                       upd_date(reg_date) { }

Account::Account(int id, const std::string& account_id, const std::string& password,
                 const std::string& reg_date, const std::string& upd_date) : id(id),
                                                                             account_id(account_id),
                                                                             password(password) { }

int Account::get_id()
{
    return id;
}

std::string Account::get_account_id() {
    return account_id;
}

std::string Account::get_password() {
    return password;
}

void Account::printAccountInfo() {

}

File: python/sungyoungjang/first/account/entity/Account.py
class Account:
    #Account::Account(std::string accountId, std::string password)
    #                                   : accountId(accountId), password(password) {~~~~~}
    def __init__(self, accountId, password):
        self.__accountId = accountId
        self.__password = password

    def getAccountId(self):
        return self.__accountId

    def setAccountId(self, accountId):
        self.__accountId = accountId


    def getPassword(self):
        return self.__password

    def setPassword(self, password):
       self.__password = password

    def __str__(self):
        return f"Account ID: {self.__accountId}, Password: {self.__password}"